plugins/kubesvc/patch-clouddriver.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: spinnaker.armory.io/v1alpha2
+kind: SpinnakerService
+metadata:
+  name: spinnaker
+spec:
+  spinnakerConfig:
+    profiles:
+      clouddriver:
+        kubernetes:
+          enabled: true
+
+        sql:
+          enabled: true
+          taskRepository:
+            enabled: true
+          cache:
+            enabled: true
+            # These parameters were determined to be optimal via benchmark comparisons
+            # in the Netflix production environment with Aurora. Setting these too low
+            # or high may negatively impact performance. These values may be sub-optimal
+            # in some environments.
+            readBatchSize: 500
+            writeBatchSize: 300
+          scheduler:
+            enabled: true
+          connectionPools:
+            default:
+              # additional connection pool parameters are available here,
+              # for more detail and to view defaults, see:
+              # https://github.com/spinnaker/kork/blob/master/kork-sql/src/main/kotlin/com/netflix/spinnaker/kork/sql/config/ConnectionPoolProperties.kt
+              default: true
+              jdbcUrl: jdbc:mysql://mysql:3306/clouddriver
+              user: clouddriver_service
+              password: encrypted:k8s!n:spin-secrets!k:mysqlCdPassword  # (Secret). Depending on db auth and how spinnaker secrets are managed
+            # The following tasks connection pool is optional. At Netflix, clouddriver
+            # instances pointed to Aurora read replicas have a tasks pool pointed at the
+            # master. Instances where the default pool is pointed to the master omit a
+            # separate tasks pool.
+            tasks:
+              user: clouddriver_service
+              password: encrypted:k8s!n:spin-secrets!k:mysqlCdPassword  # (Secret). Depending on db auth and how spinnaker secrets are managed
+              jdbcUrl: jdbc:mysql://mysql:3306/clouddriver
+          migration:
+            user: clouddriver_migrate
+            password: encrypted:k8s!n:spin-secrets!k:mysqlCdMigratePassword  # (Secret). Depending on db auth and how spinnaker secrets are managed
+            jdbcUrl: jdbc:mysql://mysql:3306/clouddriver
+
+        redis:
+          enabled: true
+          cache:
+            enabled: false
+          scheduler:
+            enabled: false
+          taskRepository:
+            enabled: false
+
+        kubesvc:
+          cluster: redis
+
+        spinnaker:
+          extensibility:
+            pluginsRootPath: /opt/clouddriver/lib/plugins
+            plugins:
+              Armory.Kubesvc:
+                enabled: true
+                extensions:
+                  armory.kubesvc:
+                    enabled: true
+
+  kustomize:
+    clouddriver:
```

Comment:
Same comment as above perhaps use:
```
patchesStrategicMerge:
  - https://armory.jfrog.io/artifactory/manifests/kubesvc-plugin/clouddriver-plugin-0.3.3.yaml
  - https://armory.jfrog.io/artifactory/manifests/kubesvc-plugin/kubesvc-plugin-config-0.3.3.yaml
```

We can definitely make a new manifest in kubesvc-plugin if it helps.

Reply:
Same comment about remote references vs local manifests

Reply:
Maybe we should stop publishing manifests to artifactory because it's confusing to not have a source of truth.

Reply:
How do we feel about creating version folders for things like kubsvc that have a plugin component? So long as we are tying plugin versions to Armory Spinnaker/OSS Spinnaker versions we'll need to have several patches available. Especially if we no longer publish to Artifactory.

Reply:
How would it look like? A thought is having branches here specific for versions of spinnaker, like a `release-2.21.x` branch that has the right version of kubesvc to use, but I'm curious about version folders

Reply:
Release branches are a neat idea, though it could potentially be confusing for users that aren't familiar with how oss/armory projects are currently built.

I'm not sure at what level you would create the structure, but something like the following makes sense to me:
```
.
├── 01_base
├── armory-2.20
├── armory-2.21
├── oss-1.20
└── oss-1.21

```

`base` would contain a lot of the things that don't change frequently between releases and then in the armory/oss directories for each version you would specify only as much as needed to change the kubesvc image and clouddriver plugin config to the appropriate versions for the release.

How we apply this more broadly for other plugin-based configs I'm not sure 🤔 I definitely see an explosion of folders/files if we're not too careful haha

Reply:
I'd prefer to leave the kubesvc version out of this - and just have them in the main kustomization as an image override:
```
images:
   - name: armory/kubesvc             
     newTag: x.y.z
```

The main issue is that for the plugin part of kubesvc, the image tag is currently in a `spec.kustomize` so using `images` won't work. 

Ideally, the operator would manage Spinnaker version <-> plugin version mapping. But that's another can of worms.